crates/turborepo-scm/src/manual.rs

```diff
@@ -30,17 +30,19 @@ pub(crate) fn hash_files(
     let mut hashes = GitHashes::new();
     for file in files.into_iter() {
         let path = root_path.resolve(file.as_ref());
-        let metadata = match path.symlink_metadata() {
-            Ok(metadata) => metadata,
-            Err(e) => {
-                if allow_missing && e.is_io_error(std::io::ErrorKind::NotFound) {
-                    continue;
+        match git_like_hash_file(&path) {
```

Comment:
So, now we just attempt to hash the file, and let `read_to_end` do its thing.

We can then investigate the returned error to see if it is `ENOENT`.